extensions/control-plane/api/management-api/contract-negotiation-api/src/test/java/org/eclipse/edc/connector/controlplane/api/management/contractnegotiation/transform/JsonObjectFromContractNegotiationTransformerTest.java

```diff
@@ -49,33 +53,56 @@ class JsonObjectFromContractNegotiationTransformerTest {
     @Test
     void transform() {
         when(context.transform(any(CallbackAddress.class), eq(JsonObject.class))).thenReturn(Json.createObjectBuilder().build());
-        var cn = ContractNegotiation.Builder.newInstance()
+        var co = ContractOffer.Builder.newInstance()
+                .id(UUID.randomUUID().toString())
+                .assetId("asset-id")
+                .policy(Policy.Builder.newInstance().build())
+                .build();
+        var cn = createContractNegotiation(co);
+
+        var jsonObject = transformer.transform(cn, context);
+
+        assertThat(jsonObject).isNotNull();
+        assertThat(jsonObject.getString(ID)).isEqualTo("test-id");
+        assertThat(jsonObject.getString(CONTRACT_NEGOTIATION_STATE)).isEqualTo(REQUESTED.name());
+        assertThat(jsonObject.getString(CONTRACT_NEGOTIATION_COUNTERPARTY_ID)).isEqualTo("counter-party-id");
+        assertThat(jsonObject.getString(CONTRACT_NEGOTIATION_COUNTERPARTY_ADDR)).isEqualTo("address");
+        assertThat(jsonObject.getString(CONTRACT_NEGOTIATION_AGREEMENT_ID)).isEqualTo("test-agreement");
+        assertThat(jsonObject.getString(CONTRACT_NEGOTIATION_NEG_TYPE)).isEqualTo("PROVIDER");
+        assertThat(jsonObject.getString(CONTRACT_NEGOTIATION_PROTOCOL)).isEqualTo("protocol");
+        assertThat(jsonObject.getString(CONTRACT_NEGOTIATION_CORRELATION_ID)).isEqualTo(cn.getCorrelationId());
+        assertThat(jsonObject.getString(CONTRACT_NEGOTIATION_ASSET_ID)).isEqualTo(co.getAssetId());
+        assertThat(jsonObject.getJsonNumber(CONTRACT_NEGOTIATION_CREATED_AT).longValue()).isEqualTo(1234);
+    }
+
+    @Test
+    void transform_withoutContractOffer() {
```

Comment:
is this case possible? I don't think a negotiation can exist without an offer, it's good to have null safety there but I don't see the reason to explicitly test such case